sm_timing/src/Timer.cpp

```diff
@@ -237,6 +237,72 @@ namespace timing {
       out << std::endl;
     }
   }
+
```

Comment:
I like the idea but this way we have a lot of duplication (the print loop body is duplicated) ... I'll quickly try to remove it.